src/xunit.analyzers/PublicMethodShouldBeMarkedAsTest.cs

```diff
@@ -42,7 +43,7 @@ internal override void AnalyzeCompilation(CompilationStartAnalysisContext compil
                     symbolContext.CancellationToken.ThrowIfCancellationRequested();
 
                     var method = (IMethodSymbol)member;
-                    if (method.MethodKind != MethodKind.Ordinary)
+                    if (method.MethodKind != MethodKind.Ordinary || method.IsStatic || method.IsAbstract)
```

Comment:
xUnit.net does support public static test methods.

Reply:
Yeah, I have tested initially and xunit failed to detect that but using xunit.analyzers.tests I got it as You said. My wrong